src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -303,5 +329,36 @@ private static bool ReportNonIOException(Exception ex)
             // catch all exception. not worth crashing VS.
             return true;
         }
+
+        private static readonly TimeSpan s_reportTimeout = TimeSpan.FromMinutes(10);
+        private static bool s_timeoutReported = false;
+
+        private static void ReportTimeout(Stopwatch watch)
+        {
+            // if we tried for 10 min and still couldn't connect. NFW some data
+            if (!s_timeoutReported && watch.Elapsed > s_reportTimeout)
+            {
+                s_timeoutReported = true;
+
+                // report service hug logs along with dump
+                WatsonReporter.Report("RequestServiceAsync Timeout", new Exception("RequestServiceAsync Timeout"), ReportDetailInfo);
+            }
+        }
+
+        private static bool s_infoBarReported = false;
```

Comment:
>private static bool s_infoBarReported = false; [](start = 8, length = 46)

Maybe it is reasonable to move class vars to the top of the class

Reply:
same as above.